blockchain/chain/src/store/chain_store.rs

```diff
@@ -135,14 +135,21 @@ where
         self.publisher
             .write()
             .await
-            .publish(HeadChange::Current(ts))
+            .publish(HeadChange::Apply(ts))
             .await;
         Ok(())
     }
 
     // subscribing returns a future sink that we can essentially iterate over using future streams
     pub async fn subscribe(&self) -> Subscriber<HeadChange> {
-        self.publisher.write().await.subscribe()
+        let sub = self.publisher.write().await.subscribe();
+        let ts = self.heaviest_tipset().await.unwrap();
+        self.publisher
+            .write()
+            .await
+            .publish(HeadChange::Current(ts))
+            .await;
```

Comment:
this seems wrong? Why would subscribing publish a headchange event?

Reply:
When we sub to head changes, the first event is resolved right away and must be the Current.

Reply:
Why can't you put this in the channel when it's created, not when it's subscribed to? You'd get unnecessary head change events every time something subscribes

Reply:
Not sure what you mean? Subscribe can be called multiple times from all over the program. And anytime we subscribe, the Subscribers first event is a current event.

Reply:
Yeah, but what I'm saying is that _all_ subscribers would receive this event, is that intended?

Reply:
Wouldn't it make sense for the subscribe fn to also pass back the current ts, instead of publishing to broadcast?

Reply:
Ohhhhhhh. Shoot, big brain oof moment. I believe youre completely right. Will make that change right now